src/code/z_view.c

```diff
@@ -1,62 +1,603 @@
+        __assert("uly >= 0", "../z_view.c", 455);
+    }
+    if (lrx > SCREEN_WIDTH) {
+        __assert("lrx <= SCREEN_WD", "../z_view.c", 456);
+    }
+    if (lry > SCREEN_HEIGHT) {
+        __assert("lry <= SCREEN_HT", "../z_view.c", 457);
+    }
+    func_800C6AC4(gfxArr, gfxCtx, "../z_view.c", 459);
+    gDPPipeSync(gfxCtx->polyOpa.p++);
+    gDPSetScissor(gfxCtx->polyOpa.p++, G_SC_NON_INTERLACE, ulx, uly, lrx, lry);
+    gDPPipeSync(gfxCtx->polyXlu.p++);
+    gDPSetScissor(gfxCtx->polyXlu.p++, G_SC_NON_INTERLACE, ulx, uly, lrx, lry);
+    func_800C6B54(gfxArr, gfxCtx, "../z_view.c", 472);
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_view/func_800AA550.s")
+#endif
+
+void func_800AA76C(View* view, f32 arg1, f32 arg2, f32 arg3) {
+    view->unk_E8.x = arg1;
+    view->unk_E8.y = arg2;
+    view->unk_E8.z = arg3;
+}
+
+void func_800AA78C(View* view, f32 arg1, f32 arg2, f32 arg3) {
+    view->unk_F4.x = arg1;
+    view->unk_F4.y = arg2;
+    view->unk_F4.z = arg3;
+}
+
+void func_800AA7AC(View* view, f32 arg1) {
+    view->unk_100 = arg1;
+}
+
+void func_800AA7B8(View* view) {
+    view->unk_E8.x = 0.0f;
+    view->unk_E8.y = 0.0f;
+    view->unk_E8.z = 0.0f;
+    view->unk_F4.x = 1.0f;
+    view->unk_F4.y = 1.0f;
+    view->unk_F4.z = 1.0f;
+    view->unk_104 = view->unk_E8;
+    view->unk_110 = view->unk_F4;
+    view->unk_100 = 0.0f;
+}
+
+void func_800AA814(View* view) {
+    view->unk_E8.x = 0.0f;
+    view->unk_E8.y = 0.0f;
+    view->unk_E8.z = 0.0f;
+    view->unk_F4.x = 1.0f;
+    view->unk_F4.y = 1.0f;
+    view->unk_F4.z = 1.0f;
+    view->unk_100 = 1.0f;
+}
+
+void func_800AA840(View* view, Vec3f vec1, Vec3f vec2, f32 arg3) {
+    view->unk_E8 = vec1;
+    view->unk_F4 = vec2;
+    view->unk_100 = arg3;
+}
+
+s32 func_800AA890(View* view, Mtx* mtx) {
+    MtxF mtxF;
+
+    if (view->unk_100 == 0.0f) {
+        return 0;
+    } else if (view->unk_100 == 1.0f) {
+        view->unk_104 = view->unk_E8;
+        view->unk_110 = view->unk_F4;
+        view->unk_100 = 0.0f;
+    } else {
+        view->unk_104.x += ((view->unk_E8.x - view->unk_104.x) * view->unk_100);
+        view->unk_104.y += ((view->unk_E8.y - view->unk_104.y) * view->unk_100);
+        view->unk_104.z += ((view->unk_E8.z - view->unk_104.z) * view->unk_100);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_view/func_800AA76C.s")
+        view->unk_110.x += ((view->unk_F4.x - view->unk_110.x) * view->unk_100);
+        view->unk_110.y += ((view->unk_F4.y - view->unk_110.y) * view->unk_100);
+        view->unk_110.z += ((view->unk_F4.z - view->unk_110.z) * view->unk_100);
+    }
+    Matrix_MtxToMtxF(mtx, &mtxF);
+    Matrix_Put(&mtxF);
+    Matrix_RotateX(view->unk_104.x, 1);
+    Matrix_RotateY(view->unk_104.y, 1);
+    Matrix_RotateZ(view->unk_104.z, 1);
+    Matrix_Scale(view->unk_110.x, view->unk_110.y, view->unk_110.z, 1);
+    Matrix_RotateZ(-view->unk_104.z, 1);
+    Matrix_RotateY(-view->unk_104.y, 1);
+    Matrix_RotateX(-view->unk_104.x, 1);
```

Comment:
Matrix_ functions that expect a "mode" argument should use values from the MatrixMode enum.